src/inventory.cpp

```diff
@@ -372,6 +372,16 @@ void inventory::restack(player *p)
     for( auto &elem : to_restack ) {
         add_item( elem );
     }
+
+    //Ensure that all items in the same stack have the same invlet.
+    for( auto outer = items.begin(); outer != items.end(); outer++ ) {
```

Comment:
Both loops could be written as range-based for loops (see loop directly above).

Also, just assign the letter, there is no need to check for it already being there.